src/server/scripts/EasternKingdoms/MagistersTerrace/boss_vexallus.cpp

```diff
@@ -15,124 +15,239 @@
 #include "magisters_terrace.h"
 
+// Boss combat text and emotes
 enum Yells
 {
-    SAY_AGGRO                       = 0,
-    SAY_ENERGY                      = 1,
-    SAY_OVERLOAD                    = 2,
-    SAY_KILL                        = 3,
-    EMOTE_DISCHARGE_ENERGY          = 4
+    SAY_AGGRO                       = 0,    // Combat start
+    SAY_ENERGY                      = 1,    // Pure energy spawn
+    SAY_OVERLOAD                    = 2,    // Final phase
+    SAY_KILL                        = 3,    // Player kill
+    EMOTE_DISCHARGE_ENERGY          = 4     // Energy discharge warning
 };
 
+// Combat ability spell IDs
 enum Spells
 {
-    // Pure energy spell info
-    SPELL_ENERGY_FEEDBACK_CHANNEL   = 44328,
-    SPELL_ENERGY_FEEDBACK           = 44335,
-
-    // Vexallus spell info
-    SPELL_CHAIN_LIGHTNING           = 44318,
-    SPELL_OVERLOAD                  = 44352,
-    SPELL_ARCANE_SHOCK              = 44319,
-
-    SPELL_SUMMON_PURE_ENERGY_N      = 44322,
-    SPELL_SUMMON_PURE_ENERGY_H1     = 46154,
-    SPELL_SUMMON_PURE_ENERGY_H2     = 46159
+    SPELL_ENERGY_FEEDBACK_CHANNEL   = 44328, // Pure energy channel
+    SPELL_ENERGY_FEEDBACK           = 44335, // Pure energy death effect
+    SPELL_CHAIN_LIGHTNING           = 44318, // Basic attack
+    SPELL_OVERLOAD                  = 44352, // Enrage ability
+    SPELL_ARCANE_SHOCK              = 44319, // Basic attack
+    SPELL_SUMMON_PURE_ENERGY_N      = 44322, // Normal mode summon
+    SPELL_SUMMON_PURE_ENERGY_H1     = 46154, // Heroic mode summon 1
+    SPELL_SUMMON_PURE_ENERGY_H2     = 46159  // Heroic mode summon 2
 };
 
-enum Misc
+// Pure energy add AI implementation
+struct npc_pure_energy : public ScriptedAI
 {
-    NPC_PURE_ENERGY                 = 24745
+    explicit npc_pure_energy(Creature* creature) : ScriptedAI(creature) {}
+
+    // Initialize on summon
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Creature* vexallus = summoner ? summoner->ToCreature() : nullptr)
+        {
+            if (Unit* target = vexallus->AI()->SelectTarget(SelectTargetMethod::Random, 0))
+            {
+                me->CastSpell(target, SPELL_ENERGY_FEEDBACK_CHANNEL, false);
+            }
+        }
+    }
+
+    // Only take damage from players targeting this add
+    void DamageTaken(Unit* attacker, uint32& damage, DamageEffectType, SpellSchoolMask) override
+    {
+        if (!attacker || !attacker->IsPlayer() || attacker->GetVictim() != me)
+            damage = 0;
+    }
+
+    // Apply feedback effect on death
+    void JustDied(Unit* killer) override
+    {
+        if (killer)
+            killer->CastSpell(killer, SPELL_ENERGY_FEEDBACK, true, nullptr, nullptr, me->GetGUID());
+    }
+
+    // Override to prevent actual melee attacks while still allowing movement
+    void AttackStart(Unit* victim) override 
+    {
+        if (victim)
+        {
+            me->Attack(victim, false); // Set victim but don't allow melee
```

Comment:
This won't prevent them from auto attacking. Override UpdateAI() instead (auto attacks are called there in DoMeleeAttackIfReady()